modules/document_repository/php/files.class.inc

```diff
@@ -0,0 +1,486 @@
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+namespace LORIS\document_repository;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+
+/**
+ * It includes two functions.
+ * Post method handles updating a file.
+ * Delete method handles deleting a file.
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Document_Repository
+ * @author   Loris Team <loris.mni@bic.mni.mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class Files extends \NDB_Page
+{
+    public $skipTemplate = true;
+
+    /**
+     * Same as the permissions for the main menu filter page.
+     *
+     * @param \User $user The user whose access is being checked
+     *
+     * @return bool true iff the user has access to this page.
+     */
+    function _hasAccess(\User $user) : bool
+    {
+        return $user->hasAnyPermission(
+            array(
+             'document_repository_view',
+             'document_repository_delete',
+            )
+        );
+    }
+
+    /**
+     * Handle how to operate all the files.
+     * PUT method updates a file's info.
+     * DELETE method deletes a file.
+     * POST method uploads a file.
+     * GET method gets a file.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        switch ($request->getMethod()) {
+        case "POST":
+            if ($this->uploadDocFile($request)) {
+                return (new \LORIS\Http\Response())
+                ->withHeader("Content-Type", "text/plain")
+                ->withStatus(200)
+                ->withBody(
+                    new \LORIS\Http\StringStream(
+                        json_encode("uploaded successfully")
+                    )
+                );
+            } else {
+                return (new \LORIS\Http\Response())
+                ->withHeader("Content-Type", "text/plain")
+                ->withStatus(200)
+                ->withBody(
+                    new \LORIS\Http\StringStream(
+                        json_encode("duplicate file name")
+                    )
+                );
+            }
+            break;
+        case "PUT":
+            $this->editDocFile();
+            return (new \LORIS\Http\Response())
+                ->withHeader("Content-Type", "text/plain")
+                ->withStatus(200)
+                ->withBody(
+                    new \LORIS\Http\StringStream(
+                        json_encode("update a file")
+                    )
+                );
+          break;
+        case "DELETE":
+            $path = $request->getQueryParams();
+            $id   = basename($path['lorispath']);
```

Comment:
This isn't reliable, the fact that `$_GET['lorispath']` is set to something is a side-effect of the implementation. You should be using `$request->getUri()->getPath()` to get the URL path. (same with the couple instances below.)